pkg/driver/constants.go

```diff
@@ -95,6 +95,10 @@ const (
 	// secretName attribute to be used to specify secret name for ephemeral inline volume
 	secretNameKey = "inline-volume-secret-name"
 
+	volEncryptionSecretNameKey = "encryptionSecretName"
+	volEncryptionSecretNamespaceKey = "encryptionSecretNamespace"
+	volEncryptionKey = "volumeEncryptionKey"
+
```

Comment:
After some debate with the PM/architects, these parameters needs to be changed into:
```
hostEncryptionSecretName
hostEncryptionSecretNamespace
```
The `Secret` key needs to be named:
```
hostEncryptionKey
```
We also need to re-introduce the boolean. This will be used for easy toggling of the host encryption feature from a PVC annotation (override during provisioning only, not mutable).

The boolean should be named:
```
hostEncryption
```

Reply:
Will rename the parameter name to have "host" prefix rather than "volume" prefix. 

As for toggling of encryption feature is concerned, once a PVC is created, be it encrypted or unencrypted, we don't support switching it to the other type.

Besides, I would like to highlight the following points with respect to the boolean hostEncryption:
1. In order to use this feature, user would need to specify an additional parameter in storage class
2. hostEncryption boolean is redundant when the presence of hostSecretName and hostSecretNamespace suffices.
3. Unnecessary validations would be required when user would be specifying one parameter while skipping the other.
4. Configurations like hostEncryption set to false while specifying hostSecrentName and hostSecretNamespace wouldn't make any sense.

In view of the above, I would suggest that we don't introduce one more parameter.

Reply:
The user experience we're trying to cater for is targeted towards two personas. The Kubernetes admin and Kubernetes user, normally locked down to certain namespaces. Kubernetes users do not have the ability to create a StorageClass as it's a cluster wide object.

I understand your concern that that the `hostEncryption` may seem redundant but it makes the user experience a bit broken if we omit it entirely. We want to cater to as many use cases as possible. So, let's discuss those as I think there's a middle road here.

- ### As a Kubernetes admin, I want to enforce `hostEncryption` for all PVCs created from my StorageClass using my host key.
  
  **StorageClass**:
  ```
  parameters:
    hostEncryptionSecretName: my-super-secret
    hostEncryptionSecretNamespace: hpe-storage
  ```
  In this case, `hostEncryption` boolean will be redundant, setting `hostEncryptionSecretName` and `hostEncryptionSecretNamespace` will automatically qualify `hostEncryption` to "true".

- ### As a Kubernetes admin, I want to allow users to selectively encrypt PVCs created from my StorageClass using my host key.

  **StorageClass**:
  ```
  parameters:
    allowOverrides: hostEncryption
    hostEncryption: "false"
    hostEncryptionSecretName: my-super-secret
    hostEncryptionSecretNamespace: hpe-storage
  ```

  **PVC**:
  ```
  metadata:
    annotations:
       csi.hpe.com/hostEncryption: "true"
  ```
  Here the `hostEncryption` boolean is critical, for two reasons. The Kubernetes user has no clue where to find the host key secret and that needs to be provided by the StorageClass. Second, without `hostEncryption`, the user would have to set the secret and secretname to empty strings and it would then allow users to specify their own host key, which is not desired, since the admin would have to allow both parameters to be overridden.
  
  The admin may use this pattern to allow users to opt-out from host encryption by using `hostEncryption` set to "true" and the annotation set to "false".

- ### As a Kubernetes admin, I want to allow users to selectively encrypt PVCs created from my StorageClass using their own host keys.

  **StorageClass**:
  ```
  parameters:
    allowOverrides: hostEncryptionSecretName,hostEncryptionSecretNamespace
  ```

  **PVC**:
  ```
  metadata:
    annotations:
       csi.hpe.com/hostEncryptionSecretName: my-secret
       csi.hpe.com/hostEncryptionSecretNamespace: my-namespace
  ```
  In this case, `hostEncryption` wouldn't be necessary as both parameters will be overridden with the users annotations.

Reply:
@datamattsson - thank you for the detailed use cases. There is a catch there though which we would need to document. For an existing PVC having user data on it, if any of the mentioned use cases is executed by the user, the data on the PVC would be wiped out.

Reply:
@imran-ansari please elaborate. The override annotations are only available to the user during provisioning of a new PVC. All parameters related to host encryption should be immutable once the PVC is provisioned. Not sure if the volume-mutator sidecar or the CSI driver has a blocklist. Anyhow, we can document it either way.

Reply:
@datamattsson  - Please ignore my previous comment. Since `allowOverride` is for new volume creation, my above observation doesn't hold good. I'll go ahead with the flag addition and push the changes again.